src/net_processing.cpp

```diff
@@ -2406,11 +2406,15 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
         if (pindex->nStatus & BLOCK_HAVE_DATA) // Nothing to do here
             return true;
 
-        if (pindex->nChainWork <= chainActive.Tip()->nChainWork || // We know something better
+        if (pindex->nChainWork <= chainActive.Tip()->nChainWork || // We know same or better
                 pindex->nTx != 0) { // We had this block at some point, but pruned it
-            if (fAlreadyInFlight) {
-                // We requested this block for some reason, but our mempool will probably be useless
+            if ((fAlreadyInFlight && blockInFlightIt->second.first == pfrom->GetId())
```

Comment:
nit: how about unpacking `blockInFlightIt->second.first` to a local like `inFlightNodeId` or the like? It's used one other place in the file and is relatively inscrutable.

Reply:
I cached the "from same peer" logic instead since it's used twice.